### ISSUE TITLE
Custom rule not visiting class method signatures

### ISSUE THREAD

**Issue by patrickreck** [palantir/tslint#3649]
### Bug Report

- __TSLint version__: 5.8
- __TypeScript version__: 2.6.2
- __Running TSLint via__: (pick one) CLI

#### Custom rule

```ts
import * as ts from "typescript";
import * as Lint from "tslint";

export class Rule extends Lint.Rules.AbstractRule {
  public static FAILURE_STRING = "Missing blank line before method";

  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
    const walker = new PaulWalker(sourceFile, this.getOptions());
    return this.applyWithWalker(walker);
  }
}

class PaulWalker extends Lint.RuleWalker {

  public visitMethodSignature(node: ts.MethodSignature) {
    console.log(node.getFullText());
  }
}
```

#### Actual behavior
Logs method signatures defined in interfaces, not methods in classes

#### Expected behavior
Log methods in interfaces and classes

I am brand new to TSLint so it's very possible that there's something I didn't get. In that case, please enlighten me.

**Comment by aervin** [comment]
@patrickreck I'd paste your code snippets into [ast-explorer](http://astexplorer.net/) to better investigate the differences between MethodDeclarations in classes and MethodSignatures in interfaces.

**Comment by JoshuaKGoldberg** [comment]
FWIW, we recommend using `ts.forEachChild` and `applyWithFunction` now instead of walkers. Closing this as non-actionable - hope you got everything to work! 😊 Please do file another issue if anything's not clear.
